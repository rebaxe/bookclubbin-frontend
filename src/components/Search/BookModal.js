import { Modal, Fade, Button } from "@material-ui/core"
import { makeStyles } from "@material-ui/core/styles"
import { ClassSharp } from "@material-ui/icons"
import CloseIcon from '@material-ui/icons/Close'
import googleImg from './images/google-attribution.png'

const useStyles = makeStyles((theme) => ({
  modal: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  paper: {
    backgroundColor: theme.palette.background.paper,
    border: '1px solid #000',
    borderRadius: 5,
    boxShadow: theme.shadows[5],
    padding: theme.spacing(2, 4, 3),
    margin: theme.spacing(2),
    position: 'relative',
    width: '80%'
  },
  closeBtn: {
    position: 'absolute',
    top: 10,
    right: 10,
    padding: 5,
  },
  bookInfoContainer: {
    display: 'flex',
    flexDirection: 'row',
    color: '#333',
    fontSize: '0.8rem',
    gap: theme.spacing(2)
  },
  googleImage: {
    height: '10px',
    padding: '10px',
    position: 'absolute',
    bottom: 10,
    right: 10
  },
}))

const BookModal = (props) => {
  const view = props.open
  const book = props.book
  const handleClose = props.handleClose
  const classes = useStyles()
  return ( 
    <Modal
        open={view}
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        className={classes.modal}
      >
      <Fade in={view} >
        <div className={classes.paper}>
          <Button className={classes.closeBtn} onClick={() => {handleClose()}}>
            <CloseIcon />
          </Button>
          {book.image && <img src={book.image} alt={`Cover for ${book.title}`}/>}
          <h2 id="transition-modal-title">{book.title}</h2>
          {book.authors && <h2 id="transition-modal-title"> by {book.authors} </h2>}
          <p id="transition-modal-description">{book.description}</p>
          <div className={classes.bookInfoContainer}>
            {book.pages && <span>Pages: {book.pages}</span>}
            {book.googleRating && <span>Google Rating: {book.googleRating}</span>}
            <img className={classes.googleImage} src={googleImg} alt="Google Attribution"/>
          </div>
        </div>
      </Fade>
    </Modal>
   );
}
 
export default BookModal;