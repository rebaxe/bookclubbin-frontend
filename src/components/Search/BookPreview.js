import { Card, CardContent, CardMedia, IconButton, Typography } from "@material-ui/core";
import { makeStyles } from '@material-ui/core/styles'
import Visibility from '@material-ui/icons/Visibility'

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 350,
    margin: '20px',
    padding: '10px'
  },
  view: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'flex-end'
  },
  btn: {
    padding: 5
  }
}))

const BookPreview = (props) => {
  const classes = useStyles()
  const book = props.book
  //const book = {
  //  title: 'Where the Crawdads Sing',
  //  authors: [ 'Delia Owens' ],
  //  description: `#1 NEW YORK TIMES BESTSELLING PHENOMENON More than 6 million copies sold A Reese Witherspoon x Hello Sunshine Book Club Pick A Business Insider Defining Book of the Decade "I can't even express how much I love this book! I didn't want this story to end!"--Reese Witherspoon "Painfully beautiful."--The New York Times Book Review For years, rumors of the "Marsh Girl" have haunted Barkley Cove, a quiet town on the North Carolina coast. So in late 1969, when handsome Chase Andrews is found dead, the locals immediately suspect Kya Clark, the so-called Marsh Girl. But Kya is not what they say. Sensitive and intelligent, she has survived for years alone in the marsh that she calls home, finding friends in the gulls and lessons in the sand. Then the time comes when she yearns to be touched and loved. When two young men from town become intrigued by her wild beauty, Kya opens herself to a new life--until the unthinkable happens. Where the Crawdads Sing is at once an exquisite ode to the natural world, a heartbreaking coming-of-age story, and a surprising tale of possible murder. Owens reminds us that we are forever shaped by the children we once were, and that we are all subject to the beautiful and violent secrets that nature keeps.`,
  //  publishedDate: '2018-08-14',
  //  pages: 384,
  //  image: 'http://books.google.com/books/content?id=CGVDDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
  //  googleRating: 4,
  //  googleId: 'CGVDDwAAQBAJ'
  //}
  return ( 
    <Card className={classes.root}>
      <div
        style={{
          display: "flex",
          alignItem: "center",
          justifyContent: "center"
        }}
      >
      <CardMedia 
      style={{
        width: "auto",
        maxHeight: "200px",
        margin: '10px'
      }}
      component="img"
      image={ book.image }
      />
      </div>
      <CardContent>
        <Typography variant="h6">{ book.title }</Typography>
        <Typography variant="h6">by { book.authors }</Typography>
        <Typography variant="body2">{ book?.description?.substring(0, 200) }...</Typography>
        <div style={{
          display: 'flex',
          flexDirection: 'row',
          justifyContent: 'center',
          gap: 10,
          marginTop: 15
        }}>
          <Typography variant="subtitle2">Rating: { book.googleRating }</Typography>
          <Typography variant="subtitle2">Pages: { book.pages }</Typography>
        </div>
        <div className={classes.view}>
          <IconButton className={classes.btn}>
            <Visibility />
          </IconButton>
        </div>
      </CardContent>
    </Card>
   );
}
 
export default BookPreview;