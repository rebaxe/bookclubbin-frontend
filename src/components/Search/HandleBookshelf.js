import {
  FormControl, InputLabel, Select, MenuItem, FormControlLabel, Checkbox,
} from '@material-ui/core'
import { Favorite, FavoriteBorder } from '@material-ui/icons'
import axios from 'axios'
import { useContext, useEffect, useState } from 'react'
import { addBook, getBookclub } from '../../api/apiCalls'
import { ClubsContext } from '../../ClubsContext'
import { UserContext } from '../../UserContext'

const HandleBookshelf = (props) => {
  const { book } = props
  const [user] = useContext(UserContext)
  const [clubs, setClubs] = useContext(ClubsContext)
  const [selectedClubId, setSelectedClubId] = useState(null)
  const [checkedRead, setCheckedRead] = useState(false)
  const [checkedTBR, setCheckedTBR] = useState(false)

  const handleChange = (e) => {
    setSelectedClubId(e.target.value)
  }

  useEffect(() => {
    setCheckedRead(false)
    setCheckedTBR(false)
    clubs.forEach((club) => {
      console.log(club)
      if (club.id === selectedClubId) {
        club.booksSaved.forEach((savedBook) => {
          console.log(savedBook.googleId)
          console.log(book.googleId)
          if (savedBook.googleId === book.googleId) {
            setCheckedTBR(true)
          }
        })
        club.booksRead.forEach((readBook) => {
          if (readBook.googleId === book.googleId) {
            setCheckedRead(true)
          }
        })
      }
    })
  }, [selectedClubId])

  const handleCheckRead = async () => {
    if (checkedTBR) {
      setCheckedTBR(false)
    }

    if (checkedRead) {
      setCheckedRead(false)
      const res = await axios({
        method: 'patch',
        url: `http://localhost:8081/api/v1/bookclubs/${selectedClubId}/books/remove`,
        data: {
          bookRead: book,
        },
      })
    } else {
      setCheckedRead(true)
      addBook(selectedClubId, 'bookRead', book)
      // const res = await axios({
      //   method: 'patch',
      //   url: `http://localhost:8081/api/v1/bookclubs/${selectedClubId}/books/add`,
      //   data: {
      //     bookRead: book,
      //   },
      // })
    }
    const updatedClubs = await axios({
      method: 'get',
      url: `${process.env.REACT_APP_GET_CLUB}/${user.id}`,
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
    })

    setClubs(updatedClubs.data)
  }

  const handleCheckTBR = async () => {
    if (checkedTBR) {
      setCheckedTBR(false)
      const res = await axios({
        method: 'patch',
        url: `http://localhost:8081/api/v1/bookclubs/${selectedClubId}/books/remove`,
        data: {
          bookSaved: book,
        },
      })
    } else {
      setCheckedTBR(true)
      const res = await axios({
        method: 'patch',
        url: `http://localhost:8081/api/v1/bookclubs/${selectedClubId}/books/add`,
        data: {
          bookSaved: book,
        },
      })
    }
    const clubData = await getBookclub(user)
    setClubs(clubData)
    // const updatedClubs = await axios({
    //   method: 'get',
    //   url: `${process.env.REACT_APP_GET_CLUB}/${user.id}`,
    //   headers: {
    //     Accept: 'application/json',
    //     'Content-Type': 'application/json',
    //   },
    // })
    // setClubs(updatedClubs.data)
  }

  return (
    <FormControl variant="outlined">
      <InputLabel id="demo-simple-select-outlined-label">Bookclub</InputLabel>
      <Select
        labelId="demo-simple-select-outlined-label"
        id="demo-simple-select-outlined"
        value={selectedClubId}
        onChange={(e) => handleChange(e)}
        label="Bookclub"
      >
        {clubs.length !== 0
          ? (
            clubs.map((club) => (
              <MenuItem value={club.id}>{club.clubname}</MenuItem>
            ))
          ) : (
            <MenuItem value="">
              <em>You&apos;re not in a bookclub yet.</em>
            </MenuItem>
          )}
      </Select>
      <FormControlLabel
        control={(
          <Checkbox
            disabled={checkedRead === true || selectedClubId === null}
            icon={<FavoriteBorder />}
            checkedIcon={<Favorite />}
            name="checkedH"
            checked={checkedTBR}
            onChange={handleCheckTBR}
          />
        )}
        label="To Be Read"
      />
      <FormControlLabel
        control={(
          <Checkbox
            disabled={selectedClubId === null}
            checked={checkedRead}
            onChange={handleCheckRead}
            name="checkedB"
            color="primary"
          />
          )}
        label="Read"
      />
    </FormControl>
  )
}

export default HandleBookshelf
