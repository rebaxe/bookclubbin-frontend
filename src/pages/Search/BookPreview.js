import {
  Card, CardContent, CardMedia, IconButton, Typography,
} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import ArrowForwardIcon from '@material-ui/icons/ArrowForward'
import { useState } from 'react'
import defaultCover from './images/defaultbookcover.png'
import BookModal from '../../components/BookModal/BookModal'

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 350,
    height: 500,
    margin: 20,
    padding: '10px',
    position: 'relative',
  },
  view: {
    width: '90%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
    position: 'absolute',
    bottom: 5,
  },
  btn: {
    padding: 5,
  },
}))

const BookPreview = (props) => {
  const classes = useStyles()
  const { book } = props
  const [openView, setOpenView] = useState(false)

  const handleViewBook = () => {
    setOpenView(true)
  }
  const handleClose = () => {
    setOpenView(false)
  }

  return (
    <div>
      <Card className={classes.root}>
        <div
          style={{
            display: 'flex',
            alignItem: 'center',
            justifyContent: 'center',
          }}
        >
          {book.image ? (
            <CardMedia
              style={{
                width: 'auto',
                height: '200px',
                margin: '10px',
              }}
              component="img"
              image={book.image}
            />
          ) : (
            <CardMedia
              style={{
                width: 'auto',
                height: '200px',
                margin: '10px',
              }}
              component="img"
              image={defaultCover}
            />
          )}
        </div>
        <CardContent>
          <Typography variant="h6">{ book.title }</Typography>
          { book.authors !== undefined && (
          <Typography variant="h6">
            by{' '}
            { book.authors }
          </Typography>
          ) }
          <Typography variant="body2">
            { book?.description?.substring(0, 100) }
            ...
          </Typography>
          <div style={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'center',
            gap: 10,
            marginTop: 15,
          }}
          />
          <div className={classes.view}>
            <IconButton className={classes.btn} onClick={handleViewBook}>
              <ArrowForwardIcon />
            </IconButton>
          </div>
        </CardContent>
      </Card>
      <BookModal open={openView} book={book} handleClose={handleClose} editable />
    </div>
  )
}

export default BookPreview
