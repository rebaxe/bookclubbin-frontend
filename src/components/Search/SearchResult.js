import BookPreview from './BookPreview.js'
import { Grid } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
  container: {
    width: '100%',
    margin: 0,
    display: 'flex',
    alignItems: 'center',
  }
}))

const SearchResult = (props) => {
  const {result: books} = props
  const classes = useStyles()
  return (
    <div>
    <Grid className={classes.container} container /*alignItems="stretch" */spacing={2} alignItems="stretch" justify="center" >
        {books.map((book) => (
          <Grid key={book.googleId} item xs={12} sm={6} md={3}>
            <BookPreview book = {book}/>
          </Grid>
        ))}
      </Grid>
    </div>
   )
}
 
export default SearchResult