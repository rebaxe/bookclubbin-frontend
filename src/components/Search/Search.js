import { TextField, Button, FormHelperText } from '@material-ui/core'
import ToggleButton from '@material-ui/lab/ToggleButton'
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup'
import SearchIcon from '@material-ui/icons/Search'
import { makeStyles } from '@material-ui/core/styles'
import googleImg from './images/google-attribution.png'
import axios from 'axios'
import { useState } from 'react'
import SearchResult from './SearchResult.js'

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1),
    },
  },
  helperText: {
    display: 'flex',
    justifyContent: 'flex-start',
    gap: theme.spacing(2)
  },
  image: {
    height: '10px',
    padding: '3px 0px'
  },
  container: {
    margin: 30
  },
  searchContainer: {
    display: 'flex',
    flexDirection: 'row',
    gap: theme.spacing(1)
  },
  searchOptions: {
    padding: '10px 5px',
    fontSize: '0.8rem',
    color: '#333'
  }
}))

const Search = () => {
  const classes = useStyles()
  const [query, setQuery] = useState(null)
  const [searchPreferences, setSearchPreferences] = useState('title')
  const [searchResult, setSearchResult] = useState(null)
  
  const handleChange = (e, newSearchPreferences) => {
    setQuery('')
    setSearchResult(null)
    setSearchPreferences(newSearchPreferences)
  }

  const handleSearch = (e) => {
    e.preventDefault()
    if (query) {
      console.log(query)
      const URL = process.env.REACT_APP_SEARCH_URL
      axios.get(URL, { 
        params: { query: `${query}+${searchPreferences}:${query}` }
      }, {
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json'
        }
      }, {
        body: JSON.stringify(query)
      }).then(res => {
        const books = res.data
        setSearchResult(books)
        console.log(searchResult)
      }).catch((error) => {
        console.log(error.message)
      })
    } else {
      console.log('Enter a query')
    }
  }

  return (
    <div className={classes.container}>
      <div className={classes.searchOptions}>
          <span>Search by </span>          
            <ToggleButtonGroup 
            value={searchPreferences}
            onChange={handleChange}
            exclusive
            size="small">
              <ToggleButton value="intitle">
                Title
              </ToggleButton>
              <ToggleButton value="inauthor">
                Author
              </ToggleButton>
             </ToggleButtonGroup>
          </div>
        <form autoComplete="off" onSubmit={handleSearch}>
        <div className={classes.searchContainer}>
          <TextField
          label="Search books"
          id="search-books"
          variant="outlined"
          fullWidth
          value={query}
          onChange={(e) => setQuery(e.target.value)}
        />
        <Button type="submit"><SearchIcon/></Button>
        </div>
        <FormHelperText className={classes.helperText}>*Search by title or author.<img className={classes.image} src={googleImg} alt="Powered by Google"></img></FormHelperText>
      </form>
      
      {searchResult !== null && searchResult.length > 0 &&
        <div>
          <SearchResult result = { searchResult }/>
        </div>
      }
    </div>
  )
}
export default Search
