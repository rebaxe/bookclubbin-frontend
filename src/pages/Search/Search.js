import {
  TextField, Button, FormHelperText, InputAdornment, Typography, CircularProgress,
} from '@material-ui/core'
import ToggleButton from '@material-ui/lab/ToggleButton'
import ToggleButtonGroup from '@material-ui/lab/ToggleButtonGroup'
import SearchIcon from '@material-ui/icons/Search'
import CloseIcon from '@material-ui/icons/Close'
import { makeStyles } from '@material-ui/core/styles'
import { useState } from 'react'
import googleImg from './images/google-attribution.png'
import SearchResult from './SearchResult'
import Error from '../../components/FlashMessages/Error'
import { searchBooks } from '../../api/apiCalls'

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1),
    },
  },
  helperText: {
    display: 'flex',
    justifyContent: 'flex-start',
    gap: theme.spacing(2),
  },
  image: {
    height: '10px',
    padding: '3px 0px',
  },
  container: {
    backgroundColor: '#f4f4f4',
    margin: 30,
    padding: '2rem',
    borderRadius: 20,
  },
  searchContainer: {
    display: 'flex',
    flexDirection: 'row',
    gap: theme.spacing(1),
  },
  searchOptions: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: theme.spacing(1),
    padding: '0px 5px 10px',
    fontSize: '0.8rem',
  },
  searchField: {
    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: 'green',
    },
    '&:hover .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: 'red',
    },
    '& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline': {
      borderColor: 'purple',
    },
  },
  loading: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
}))

const Search = () => {
  const classes = useStyles()
  const [query, setQuery] = useState('')
  const [searchPreferences, setSearchPreferences] = useState('intitle')
  const [searchResult, setSearchResult] = useState(null)
  const [openError, setOpenError] = useState(false)
  const [isLoading, setIsLoading] = useState(false)

  const handleChange = (e, newSearchPreferences) => {
    setQuery('')
    setOpenError(false)
    setSearchResult(null)
    setSearchPreferences(newSearchPreferences)
  }

  const handleSearch = async (e) => {
    setIsLoading(true)
    setOpenError(false)
    e.preventDefault()
    if (query) {
      const result = await searchBooks(query, searchPreferences)

      if (result.status !== 200) {
        toggleError()
      }
      setSearchResult(result.data)
      setIsLoading(false)
    }
  }

  const toggleError = () => {
    openError ? setOpenError(false) : setOpenError(true)
  }

  return (
    <div className={classes.container}>
      <div className={classes.searchOptions}>
        <Typography variant="subtitle2">Search by </Typography>
        <ToggleButtonGroup
          value={searchPreferences}
          onChange={handleChange}
          exclusive
          size="small"
        >
          <ToggleButton value="intitle">
            Title
          </ToggleButton>
          <ToggleButton value="inauthor">
            Author
          </ToggleButton>
        </ToggleButtonGroup>
      </div>
      <form autoComplete="off" onSubmit={handleSearch}>
        <div className={classes.searchContainer}>
          <TextField
            className={classes.searcField}
            label="Search books"
            id="search-books"
            variant="outlined"
            fullWidth
            required
            value={query}
            onChange={(e) => setQuery(e.target.value)}
            InputProps={{
              endAdornment: <InputAdornment position="end">{query && <Button onClick={() => setQuery('')}><CloseIcon /></Button>}</InputAdornment>,
            }}
          />
          <Button type="submit"><SearchIcon /></Button>
        </div>
        <FormHelperText className={classes.helperText}>
          *Search by title or author.
          <img className={classes.image} src={googleImg} alt="Powered by Google" />
        </FormHelperText>
      </form>
      {searchResult !== null && searchResult.length > 0
        && (
        <div>
          <SearchResult result={searchResult} />
        </div>
        )}
      {openError && <Error open={openError} toggleError={toggleError} message="No results found." />}
      {isLoading && (
        <div className={classes.loading}>
          <CircularProgress />
        </div>
      )}
    </div>
  )
}
export default Search
