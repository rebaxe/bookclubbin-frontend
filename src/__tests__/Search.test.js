import { shallow } from 'enzyme'
import Search from '../components/Search/Search'

describe('<Search />', () => {
  const search = shallow(<Search />)
  it('should match the snapshot', () => {
    expect(search.html()).toMatchSnapshot()
  })
})
