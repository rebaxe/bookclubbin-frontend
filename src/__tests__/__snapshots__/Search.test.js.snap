// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match the snapshot 1`] = `"<div class=\\"makeStyles-container-4\\"><div class=\\"makeStyles-searchOptions-6\\"><h6 class=\\"MuiTypography-root MuiTypography-subtitle2\\">Search by </h6><div role=\\"group\\" class=\\"MuiToggleButtonGroup-root\\"><button class=\\"MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal Mui-selected MuiToggleButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" value=\\"intitle\\" aria-pressed=\\"true\\"><span class=\\"MuiToggleButton-label\\">Title</span></button><button class=\\"MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-sizeSmall\\" tabindex=\\"0\\" type=\\"button\\" value=\\"inauthor\\" aria-pressed=\\"false\\"><span class=\\"MuiToggleButton-label\\">Author</span></button></div></div><form autoComplete=\\"off\\"><div class=\\"makeStyles-searchContainer-5\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required\\" data-shrink=\\"false\\" for=\\"search-books\\" id=\\"search-books-label\\">Search books<span aria-hidden=\\"true\\" class=\\"MuiFormLabel-asterisk MuiInputLabel-asterisk\\"> *</span></label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"search-books\\" required=\\"\\" value=\\"\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\"/><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\"></p></div><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-11\\"><span>Search books *</span></legend></fieldset></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></span></button></div><p class=\\"MuiFormHelperText-root makeStyles-helperText-2\\">*Search by title or author.<img class=\\"makeStyles-image-3\\" src=\\"google-attribution.png\\" alt=\\"Powered by Google\\"/></p></form></div>"`;
